src/shims/unix/foreign_items.rs

```diff
@@ -292,6 +292,13 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                 this.write_scalar(result, dest)?;
             }
             "pipe2" => {
+                // Currently this function does not exist on all Unixes, e.g. on macOS.
+                if !matches!(&*this.tcx.sess.target.os, "linux") {
```

Comment:
Looking at the libc crate, it seems to exist on freebsd, solaris, illumos as well. Can you add those here and in the test?